### PR TITLE
Validate database

### PR DESCRIPTION
**Pull request by phornsopheak**

### CODE REVIEW

File: app/models/course.rb [hunk 1]
```diff
@@ -3,4 +3,7 @@ class Course < ActiveRecord::Base
   has_many :users, through: :user_courses
   has_many :course_subjects, dependent: :destroy
   has_many :subjects, through: :course_subjects
+  validates :name,  presence: true, length: { maximum: 50 }
```
Comment (dieunb) [inline app/models/course.rb]: add new white line above this to slice some block state code. Coding convention, pls follow at https://github.com/framgia/coding-standards/blob/master/vn/README.md

File: app/models/course.rb [hunk 2]
```diff
@@ -3,4 +3,7 @@ class Course < ActiveRecord::Base
...
+  validates :details,  presence: true, length: { maximum: 500 }
```
Comment (dieunb) [inline app/models/course.rb]: coding convention

File: app/models/course.rb [hunk 3]
```diff
@@ -3,4 +3,7 @@ class Course < ActiveRecord::Base
...
+  validates :status, presence: true, length: { maximum: 10 }
```
Comment (dieunb) [inline app/models/course.rb]: coding convention

File: app/models/course_subject.rb [hunk 4]
```diff
@@ -1,4 +1,6 @@
 class CourseSubject < ActiveRecord::Base
   belongs_to :course
   belongs_to :subject
+  validates :course_id, presence: true
```
Comment (dieunb) [inline app/models/course_subject.rb]: add new white line above this

File: app/models/subject.rb [hunk 5]
```diff
@@ -4,4 +4,7 @@ class Subject < ActiveRecord::Base
   has_many :courses, through: :course_subjects
   has_many :user_subjects, dependent: :destroy
   has_many :users, through: :user_subjects
+  validates :name, presence: true, length: { maximum: 50 }
```
Comment (dieunb) [inline app/models/subject.rb]: add new white line above this

File: app/models/subject.rb [hunk 6]
```diff
@@ -4,4 +4,7 @@ class Subject < ActiveRecord::Base
...
+  validates :details, presence: true, length: { maximum: 500 }
```
Comment (dieunb) [inline app/models/subject.rb]: Coding convention

File: app/models/subject.rb [hunk 7]
```diff
@@ -4,4 +4,7 @@ class Subject < ActiveRecord::Base
...
+  validates :status, presence: true, length: { maximum: 10 }
```
Comment (dieunb) [inline app/models/subject.rb]: Coding convention

File: app/models/task.rb [hunk 8]
```diff
@@ -2,4 +2,7 @@ class Task < ActiveRecord::Base
   belongs_to :subject
   has_many :user_tasks, dependent: :destroy
   has_many :users, through: :user_tasks
+  validates :name, presence: true, length: { maximum: 50 }
```
Comment (dieunb) [inline app/models/task.rb]: following above comment. Coding convention

File: app/models/task.rb [hunk 9]
```diff
@@ -2,4 +2,7 @@ class Task < ActiveRecord::Base
...
+  validates :status, presence: true, length: { maximum: 10 }
```
Comment (dieunb) [inline app/models/task.rb]: Coding convention

File: app/models/user.rb [hunk 10]
```diff
@@ -6,4 +6,11 @@ class User < ActiveRecord::Base
   has_many :subjects, through: :user_subjects
   has_many :user_tasks, dependent: :destroy
   has_many :tasks, through: :user_tasks
+  validates :name,  presence: true, length: { maximum: 50 }
```
Comment (dieunb) [inline app/models/user.rb]: Coding convention

File: app/models/user.rb [hunk 11]
```diff
@@ -6,4 +6,11 @@ class User < ActiveRecord::Base
...
+  before_save { self.email = email.downcase }
```
Comment (dieunb) [inline app/models/user.rb]: Coding convention

File: app/models/user.rb [hunk 12]
```diff
@@ -6,4 +6,11 @@ class User < ActiveRecord::Base
...
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: { maximum: 255 },
```
Comment (dieunb) [inline app/models/user.rb]: Coding convention

File: app/models/user.rb [hunk 13]
```diff
@@ -6,4 +6,11 @@ class User < ActiveRecord::Base
...
+                    format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: { case_sensitive: false }
+  validates :password, length: { minimum: 6 }
```
Comment (dieunb) [inline app/models/user.rb]: Coding convention

File: app/models/user.rb [hunk 14]
```diff
@@ -6,4 +6,11 @@ class User < ActiveRecord::Base
   has_many :subjects, through: :user_subjects
   has_many :user_tasks, dependent: :destroy
   has_many :tasks, through: :user_tasks
+  validates :name,  presence: true, length: { maximum: 50 }
+  before_save { self.email = email.downcase }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: { maximum: 255 },
+                    format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: { case_sensitive: false }
```
Comment (dieunb) [inline app/models/user.rb]: Coding convention

File: app/models/user.rb [hunk 15]
```diff
@@ -6,4 +6,11 @@ class User < ActiveRecord::Base
   has_many :subjects, through: :user_subjects
   has_many :user_tasks, dependent: :destroy
   has_many :tasks, through: :user_tasks
+  validates :name,  presence: true, length: { maximum: 50 }
+  before_save { self.email = email.downcase }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: { maximum: 255 },
+                    format: { with: VALID_EMAIL_REGEX },
```
Comment (dieunb) [inline app/models/user.rb]: Coding convention

File: app/models/user_course.rb [hunk 16]
```diff
@@ -1,4 +1,6 @@
 class UserCourse < ActiveRecord::Base
   belongs_to :user
   belongs_to :course
+  validates :user_id, presence: true
```
Comment (dieunb) [inline app/models/user_course.rb]: add a new white line above this

File: app/models/user_subject.rb [hunk 17]
```diff
@@ -1,3 +1,6 @@
 class UserSubject < ActiveRecord::Base
   belongs_to :subject
+  belongs_to :user
+  validates :user_id, presence: true
```
Comment (dieunb) [inline app/models/user_subject.rb]: following above comment

File: app/models/user_task.rb [hunk 18]
```diff
@@ -1,4 +1,6 @@
 class UserTask < ActiveRecord::Base
   belongs_to :user
   belongs_to :task
+  validates :user_id, presence: true
```
Comment (dieunb) [inline app/models/user_task.rb]: following above comment

**Comment by limkimhuor** [comment]
OK

**Comment by samnangcattor** [comment]
Ok

**Comment by limkimhuor** [comment]
Ok

File: app/models/activity.rb [hunk 19]
```diff
@@ -1,3 +1,7 @@
 class Activity < ActiveRecord::Base
   belongs_to :user
+
+  validates :user_id, presence: true
+  validates :id_target, presence: true
```
Comment (dieunb) [inline app/models/activity.rb]: `target_id`

File: app/models/activity.rb [hunk 20]
```diff
@@ -1,3 +1,7 @@
...
+  validates :state_target, presence: true, length: {maximum: 50}
```
Comment (dieunb) [inline app/models/activity.rb]: simple is `state`

**Comment by RathanakSreang** [comment]
Ok

File: app/models/user.rb [hunk 21]
```diff
@@ -6,4 +6,12 @@ class User < ActiveRecord::Base
   has_many :subjects, through: :user_subjects
   has_many :user_tasks, dependent: :destroy
   has_many :tasks, through: :user_tasks
+
+  validates :name,  presence: true, length: {maximum: 50}
+  before_save {self.email = email.downcase}
```
Comment (dieunb) [inline app/models/user.rb]: it is callback, you should move to other block code.

**Comment by samnangcattor** [comment]
Ok

**Pull request merged**